clang/lib/Driver/Driver.cpp

```diff
@@ -4605,41 +4605,27 @@ class OffloadingActionBuilder final {
         !InputArg->getOption().hasFlag(options::LinkerInput) &&
         !types::isSrcFile(HostAction->getType())) {
       std::string InputName = InputArg->getAsString(Args);
-      // Do not create an unbundling action for an object when we know a fat
-      // static library is being used.  A separate unbundling action is created
-      // for all objects and the fat static library.
-      // But in MSVC environment static offload archives are handled differently
-      // due to absence of partial linking support in the linker. Instead of
-      // partially linking input objects and static archives and then unbundling
-      // result we are unbundling all objects and offload archives to extract
-      // device parts. Therefore, in case on MSVC environment unbundling action
-      // for objects is still needed.
-      if (C.getDefaultToolChain().getTriple().isWindowsMSVCEnvironment() ||
+      ActionList HostActionList;
+      Action *A(HostAction);
+      // Only check for FPGA device information when using fpga SubArch.
+      if (Args.hasArg(options::OPT_fintelfpga) &&
```

Comment:
Maybe a triple check could be made instead?

Reply:
I'll be doing a larger overarching `-fintelfpga` -> FPGA triple change sometime in the near future which will involve modifying items like this.  I'll address this then.